src/main/java/net/neoforged/neoforge/event/tick/EntityTickEvent.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.tick;
+
+import net.minecraft.world.entity.Entity;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.neoforge.event.entity.EntityEvent;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Base class of the two entity tick events.
+ * 
+ * @see Pre
+ * @see Post
+ */
+public abstract class EntityTickEvent extends EntityEvent {
+    @ApiStatus.Internal
```

Comment:
Nit: This is inconsistent with the other events. We probably want to make the constructor protected too.